src/group/tests/kat_transcripts.rs

```diff
@@ -0,0 +1,219 @@
+//! # Known Answer Tests for the transcript hashes
+//!
+//! See https://github.com/mlswg/mls-implementations/blob/master/test-vectors.md
+//! for more description on the test vectors.
+
+use std::convert::TryFrom;
+
+use crate::{
+    ciphersuite::{CiphersuiteName, Secret, Signature},
+    codec::Codec,
+    config::Config,
+    group::{
+        update_confirmed_transcript_hash, update_interim_transcript_hash, GroupContext, GroupEpoch,
+        GroupId,
+    },
+    messages::{Commit, ConfirmationTag},
+    prelude::{
+        random_u32, random_u64, randombytes, sender::SenderType, ContentType, LeafIndex,
+        MLSPlaintext, MLSPlaintextCommitAuthData, MLSPlaintextCommitContent,
+        MLSPlaintextContentType, Sender,
+    },
+    schedule::{ConfirmationKey, MembershipKey},
+    test_util::{bytes_to_hex, hex_to_bytes, read, write},
+};
+
+use serde::{self, Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+struct TranscriptTestVector {
+    cipher_suite: u16,
+    group_id: String,
+    epoch: u64,
+    tree_hash_before: String,
+    confirmed_transcript_hash_before: String,
+    interim_transcript_hash_before: String,
+    membership_key: String,
+    confirmation_key: String,
+    commit: String, // TLS serialized MLSPlaintext(Commit)
+
+    confirmed_transcript_hash_after: String,
+    interim_transcript_hash_after: String,
+}
+
+#[test]
+fn generate_test_vectors() {
+    let mut tests = Vec::new();
+    const NUM_TESTS: usize = 100;
+
+    for ciphersuite in Config::supported_ciphersuites() {
+        for _ in 0..NUM_TESTS {
+            // Generate random values.
+            let group_id = GroupId::random();
+            let epoch = random_u64();
+            let tree_hash_before = randombytes(ciphersuite.hash_length());
+            let confirmed_transcript_hash_before = randombytes(ciphersuite.hash_length());
+            let interim_transcript_hash_before = randombytes(ciphersuite.hash_length());
+            let membership_key =
+                MembershipKey::from_secret(Secret::random(ciphersuite.hash_length()));
+            let confirmation_key =
+                ConfirmationKey::from_secret(Secret::random(ciphersuite.hash_length()));
+
+            // Build plaintext commit message.
+            let mut commit = MLSPlaintext {
+                group_id: GroupId::random(),
+                epoch: GroupEpoch(epoch),
+                sender: Sender {
+                    sender_type: SenderType::Member,
+                    sender: LeafIndex::from(random_u32()),
+                },
+                authenticated_data: randombytes(48),
+                content_type: ContentType::Commit,
+                content: MLSPlaintextContentType::Commit(Commit {
+                    proposals: vec![],
+                    path: None,
+                }),
+                signature: Signature::new_empty(),
+                confirmation_tag: None,
+                membership_tag: None,
+            };
```

Comment:
This probably doesn't matter here, but this Commit message wouldn't be valid because it should either contain a `path` or add proposals.

Reply:
yeah we don't care about validity here